crates/ty_python_semantic/src/types.rs

```diff
@@ -5660,6 +5735,79 @@ impl<'db> From<&Type<'db>> for Type<'db> {
     }
 }
 
+impl<'db> VarianceInferable<'db> for Type<'db> {
+    fn variance_of(self, db: &'db dyn Db, type_var: TypeVarInstance) -> TypeVarVariance {
+        tracing::debug!(
+            "Checking variance of '{tvar}' in `{ty:?}`",
+            tvar = type_var.name(db),
+            ty = self,
+        );
```

Comment:
There's a helper method for printing out a nicer rendering of the type:

```suggestion
        tracing::debug!(
            "Checking variance of '{tvar}' in `{ty}`",
            tvar = type_var.name(db),
            ty = self.display(db),
        );
```